components/net/bluetooth_test.rs

```diff
@@ -0,0 +1,545 @@
+                                     CHARACTERISTIC_USER_DESCRIPTION_UUID.to_owned(),
+                                     vec![22, 33, 44, 55],
+                                     sender);
+
+    // Client Characteristic Configuration descriptor
+    let _client_characteristic_configuration =
+        create_descriptor_with_value(&device_name_characteristic,
+                                     CLIENT_CHARACTERISTIC_CONFIGURATION_UUID.to_owned(),
+                                     vec![0, 0],
+                                     sender);
+
+    // Peripheral Privacy Flag Characteristic
+    let peripheral_privacy_flag_characteristic =
+        create_characteristic(&generic_access_service,
+                              PERIPHERAL_PRIVACY_FLAG_CHARACTERISTIC_UUID.to_owned(),
+                              sender);
+    set_attribute_or_return_error(peripheral_privacy_flag_characteristic
+                                  .set_flags(vec![READ_FLAG.to_string(), WRITE_FLAG.to_string()]),
+                                  sender);
+    generic_access_service
+}
+
+// Create Heart Rate Device
+fn create_heart_rate_device(adapter: &BluetoothAdapter,
+                            sender: &IpcSender<BluetoothResult<()>>)
+                            -> BluetoothDevice {
+    // Heart Rate Device
+    let heart_rate_device =
+        create_device_with_uuids(adapter,
+                                 HEART_RATE_DEVICE_NAME.to_owned(),
+                                 HEART_RATE_DEVICE_ADDRESS.to_owned(),
+                                 vec![GENERIC_ACCESS_SERVICE_UUID.to_owned(), HEART_RATE_SERVICE_UUID.to_owned()],
+                                 sender);
+
+    // Generic Access Service
+    let _generic_access_service = create_generic_access_service(&heart_rate_device, sender);
+
+    // Heart Rate Service
+    let _heart_rate_service = create_heart_rate_service(&heart_rate_device, sender);
+
+    heart_rate_device
+}
+
+pub fn test(manager: &mut BluetoothManager, data_set_name: String, sender: &IpcSender<BluetoothResult<()>>) {
+    match manager.get_or_create_adapter().as_ref() {
+        Some(adapter) => {
+            match data_set_name.as_str() {
+                NOT_PRESENT_ADAPTER => {
+                    set_adapter(adapter, NOT_PRESENT_ADAPTER.to_owned(), sender);
+                    set_attribute_or_return_error(adapter.set_present(false), sender);
+                },
+                NOT_POWERED_ADAPTER => {
+                    set_adapter(adapter, NOT_POWERED_ADAPTER.to_owned(), sender);
+                    set_attribute_or_return_error(adapter.set_powered(false), sender);
+                },
+                EMPTY_ADAPTER => {
+                    set_adapter(adapter, EMPTY_ADAPTER.to_owned(), sender);
+                },
+                GLUCOSE_HEART_RATE_ADAPTER => {
+                    set_adapter(adapter, GLUCOSE_HEART_RATE_ADAPTER.to_owned(), sender);
+
+                    // Glucose Device
```

Comment:
I don't think the comments with the device names in this code add much - we already call the local variables names like `glucose_device`, and we're passing arguments like `GLUCOSE_DEVICE_NAME` :)